Tutorials/whatiswot/12-Building_Blocks/12-Building-Blocks.vtt

```diff
@@ -0,0 +1,197 @@
+
+00:02:45.377 --> 00:02:51.646 
+They are not limited to a local area network but they also support remote discovery.
+
+00:02:51.646 --> 00:02:57.684 
+To achieve these functionalities, WoT Discovery uses a two-stage process.
+
+00:02:57.684 --> 00:03:02.727 
+In the first stage, an introduction is made with one of
+
+00:03:02.727 --> 00:03:07.771 
+the several first-contact mechanisms such as QR codes, mDNS, and DIDs.
+
+00:03:07.771 --> 00:03:12.152 
+The second stage is the exploration phase where TDs are made available
+
+00:03:12.152 --> 00:03:16.534 
+in TD directories so that they can be managed and searched.
+
+00:03:16.534 --> 00:03:26.646 
+After the introduction, WoT Discovery defines a set of second-stage "exploration" mechanisms that provide actual access to metadata.
+
+00:03:26.646 --> 00:03:30.634 
+This can be done in 2 ways: a single TD can
+
+00:03:30.634 --> 00:03:34.621 
+be retrieved or a link for TD directories becomes available.
+
+00:03:34.621 --> 00:03:39.596 
+It is a database with a REST API to access TDs.
+
+00:03:39.596 --> 00:03:50.171 
+Thing Description Directory service provides mechanisms to register, and retrieve WoT metadata and TDs after authentication and authorization.
+
+00:03:50.171 --> 00:03:56.059 
+Thus we can summarize the Discovery mechanism like this.
+
+00:03:56.059 --> 00:04:00.021 
+The third building block is WoT Binding Templates.
+
+00:04:00.021 --> 00:04:08.290 
+The IoT uses a variety of protocols for accessing devices because no single protocol is enough to cover the different
+
+00:04:08.290 --> 00:04:16.559 
+use cases and requirements. The challenge of the WoT is to enable interactions of a variety of different IoT platforms.
+
+00:04:16.559 --> 00:04:23.016 
+The Binding Templates address an application client to use the TD so that they can extract
+
+00:04:23.016 --> 00:04:29.474 
+metadata of the protocols such as HTTP and MQTT etc., payload formats such as JSON and
+
+00:04:29.474 --> 00:04:35.931 
+CBOR, and their usage in an IoT framework context such as ECHONET and OPC.
+
+00:04:35.931 --> 00:04:42.403 
+In general, Binding Templates provides a collection of blueprints that give guidance on how to
+
+00:04:42.403 --> 00:04:48.874 
+interact with different Things that use different transport protocols, and content types or that are
+
+00:04:48.874 --> 00:04:55.346 
+different IoT platforms or standards that use certain combinations of transport protocols and content types.
+
+00:04:55.346 --> 00:04:58.996 
+Now we will introduce the WoT profile.
+
+00:04:58.996 --> 00:05:06.183 
+By nature, TD is flexible and able to describe anything without any limitations.
+
+00:05:06.183 --> 00:05:14.940 
+However, in some cases let's say you are trying to communicate with your smart bulb with your smartphone,
+
+00:05:14.940 --> 00:05:23.696 
+the lack of limitation of the TDs causes an unnecessary implementation load to communicate with every Thing.
```

Comment:
The animation timing is off. The reduction part of the animation should happen in the next subtitle